src/components/Player/Profile.svelte

```diff
@@ -59,19 +63,74 @@
 	}
 
 	let editError = null;
-	function onPlayerDataEditError(err) {
-		editError = err?.detail ?? null;
-	}
 
 	let roles = null;
 	function updateRoles(role) {
-		roles = role?.split(',').reverse();
+		roles =
+			role
+				?.split(',')
+				?.reverse()
+				?.filter(r => r?.length) ?? [];
 	}
 
 	function refreshPinnedScores(pinnedScores) {
 		$pinnedScoresStore = pinnedScores ?? [];
 	}
 
+	function onEnableEditModel() {
+		editModel = editModel = {
+			data: {
+				name: playerData?.name ?? '',
+				country: playerData?.playerInfo?.countries?.[0]?.country?.toLowerCase() ?? '',
+				avatar: null,
+				patreonMessage: playerData?.playerInfo?.patreonFeatures?.message ?? '',
```

Comment:
I moved this to a `profileSettings` as well, please use only them so I can delete `patreonFeatures` in future